src/dataStructures.hpp

```diff
@@ -205,33 +205,158 @@ struct SpongeZoneData {
 
-  // used in gradient computation:
-  // gradients of shape functions for all nodes and weight multiplied by det(Jac)
-  // at each integration point
-  Vector elemShapeDshapeWJ;           // [...l_0(i),...,l_dof(i),l_0_x(i),...,l_dof_d(i), w_i*detJac_i ...]
-  Array<int> elemPosQ_shapeDshapeWJ;  // position and num. of integration points for each element
+  /** for each interior face, index of element 1 */
+  Array<int> face_el1;
+
+  /** for each interior face, index of element 2 */
+  Array<int> face_el2;
+
+  /** for each interior face, number of quadrature points */
+  Array<int> face_num_quad;
+};
+
+/** @brief Data for boundary face integral calculations
+ *
+ * The _GPU_ path requires pre-computation of a number of quantities,
+ * such as the values of the shape functions at each quadrature point
+ * on each face.  For the boundary faces, these quantities, which are
+ * documented in more detail below, are stored in this struct.
+ *
+ */
+struct boundaryFaceIntegrationData {
+  /** Shape functions evaluated at all boundary face quadrature points */
+  Vector face_shape;
+
+  /** Weights associated with all boundary face quadrature points */
+  Vector face_quad_weight;
+
+  /** Normal vector (outward pointing) for all boundary face quadrature points */
+  Vector face_normal;
+
+  /** for each boundary face, index of corresponding element */
+  Array<int> face_el;
+
+  /** for each boundary face, number of quadrature points */
+  Array<int> face_num_quad;
 };
 
-struct parallelFacesIntegrationArrays {
-  Vector sharedShapeWnor1;
-  Vector sharedShape2;
-  Array<int> sharedElem1Dof12Q;
-  Array<int> sharedVdofs;
-  Array<int> sharedVdofsGradUp;
+struct sharedFaceIntegrationData {
+  /** Shape functions for "element 1" evaluated at all shared face quadrature points */
+  Vector face_el1_shape;
+
+  /** Shape functions for "element 2" evaluated at all shared face quadrature points */
+  Vector face_el2_shape;
+
+  /** Weights associated with all shared face quadrature points */
+  Vector face_quad_weight;
+
+  /** Normal vector (oriented from elem 1 toward elem 2) for all shared face quadrature points */
+  Vector face_normal;
+
+  /** for each shared face, index of element 1 */
+  Array<int> face_el1;
+
+  /** for each shared face, number of dofs for element 2 */
+  Array<int> face_num_quad;
+
+  /** for each shared face, number of quadrature points */
+  Array<int> face_num_dof2;
+
+  /** @brief Map from shared element index to shared face index(ices)
+   *
+   *  This mapping makes it possible to loop over the elements with
+   *  shared faces rather than the shared faces themselves.  It
+   *  contains the following information:
+   *
+   * sharedElemsFaces[7*i] = element number for ith element with shared faces
+   * sharedElemsFaces[7*i + 1] = number of shared faces on ith elem with shared faces
+   * sharedElemsFaces[7*i + 1 + j] = shared face index for the jth shared face on the ith element with shared faces
+   */
```

Comment:
change name for style consistency (and think of more descriptive name if possible)

Reply:
7c388b4